engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala

```diff
@@ -11,19 +15,28 @@ sealed trait LoggedUser {
 }
 
 object LoggedUser {
-   def apply(id: String,
-             username: String,
-             categoryPermissions: Map[String, Set[Permission]] = Map.empty,
-             globalPermissions: List[GlobalPermission] = Nil,
-             isAdmin: Boolean = false): LoggedUser = {
-     if (isAdmin) {
-       AdminUser(id, username)
-     } else {
-       CommonUser(id, username, categoryPermissions, globalPermissions)
-     }
+  def apply(authenticatedUser: AuthenticatedUser, config: Config, processCategories: List[String]): LoggedUser = {
+    apply(authenticatedUser, AuthenticationConfiguration.getRules(config), processCategories)
```

Comment:
we read config with each invocation here. Could we do this only once at application start?

Reply:
I have moved config reading to the main application class.

Reply:
A separate commit so you can see the difference.